example/main.rs

```diff
@@ -42,6 +43,31 @@ fn print_response<T: ApiResult>(response: ApiResponse<T>) {
     }
 }
 
+/// Sometimes you want to pipe results to jq etc
```

Comment:
Can we also print out errors in JSON format? Right now, this function's output can only be piped out to `jq` if the response is a success. The error messages from the API should also be json-compatible.

Reply:
i tried to but adding serialize impls to the API error structs was significantly more overhead, and `example/main` is for manual testing anyway, so if errors break pipes i'm not personally too bothered.

Reply:
Huh! I would have thought serde would have offered a nice default implementation. (`#derive[default]`)

Reply:
i can give it a try; i tried a derive deserialize

Reply:
We'll shelve this for now :)